ip-utils.c

```diff
@@ -191,10 +223,32 @@ int janus_network_address_string_buffer_is_null(const janus_network_address_stri
 	return !b || b->family == AF_UNSPEC;
 }
 
-const char * janus_network_address_string_from_buffer(const janus_network_address_string_buffer *b) {
+const char *janus_network_address_string_from_buffer(const janus_network_address_string_buffer *b) {
 	if(janus_network_address_string_buffer_is_null(b)) {
 		return NULL;
 	} else {
 		return b->family == AF_INET ? b->ipv4 : b->ipv6;
 	}
 }
+
+int janus_network_lookup_interface(const struct ifaddrs *ifas, const char *iface, janus_network_address *result) {
+	if(result == NULL)
+		return -1;
+	janus_network_address_nullify(result);
+	if(iface == NULL)
+		return -1;
+	janus_network_query_config q;
+	/* Let's see if iface is an IPv4 address, an IPv6 address, or possibly an interface name */
+	if(janus_network_prepare_device_query(iface,
+			janus_network_query_options_ipv4 | janus_network_query_options_ipv6 | janus_network_query_options_name, &q)) {
+		/* None of them..? */
+		return -1;
```

Comment:
That goes for other places where a simple `return -1` is performed: it is better to use `- <errno>` type error codes with values from `errno.h`. That way the caller can inspect the return code and take more informed decisions based on the type of error being signaled, even if that is something trivial like:
```
    int error = janus_network_lookup_interface(...);
    if(error) {
        JANUS_LOG(LOG_ERR, "Oh noes: %s\n", strerror(-error));
    }
```